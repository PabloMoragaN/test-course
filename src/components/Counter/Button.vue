<template lang="pug">
  p {{ count }}
  button(@click="handleClick") Incrementar valor
</template>

<script>
export default {
  name: "CButton",

  props: {
    initialValue: {
      type: Number,
      default: 0
    },
  },

  data() {
    return {
      counter: 0
    }
  },

  methods: {
    handleClick () {
      this.counter++
      this.$emit('count', this.count)
    }
  }
}
</script>

<style lang="scss" scoped>
  .c-button {
    &--medium {
      font-size: 1rem;
    }
    &--big {
      font-size: 2rem;
    }
    &--small {
      font-size: .5rem;
    }
  }
</style>
