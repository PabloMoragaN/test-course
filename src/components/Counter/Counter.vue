<template lang="pug">
  div
    p(ref="count") {{ count }}
    button(@click="handleClick", ref="button") Incrementar valor
</template>

<script>
export default {
  name: "counter",

  props: {
    initialValue: {
      type: Number,
      default: 0,
      validator: value => value >= 0
    },
  },

  data() {
    return {
      count: 0
    }
  },

  mounted () {
    this.count = this.initialValue
  },

  methods: {
    handleClick () {
      this.count++
      this.$emit('count', this.count)
    }
  }
}
</script>
