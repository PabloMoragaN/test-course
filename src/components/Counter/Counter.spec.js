import { shallowMount } from '@vue/test-utils'
import Counter from './Counter.vue'

describe('Counter test suite', () => {

  const sampleInitialValue = 4
  describe(`:initialValue | Cuando se pasa ${sampleInitialValue} como valor inicial`, () => {
    const wrapper = shallowMount(Counter, { propsData: { initialValue: sampleInitialValue } })
    it(`El valor del contador es ${sampleInitialValue}`, () => {
      expect(wrapper.vm.count).toBe(sampleInitialValue)
    })
    it('El contador incrementa su valor en 1', () => {
      const p = wrapper.find({ ref: 'count' }) // Buscamos el elemento donde debe pintarse
      expect(Number(p.text())).toBe(sampleInitialValue) // Comprobamos que el texto coincide
    })
  })

  describe(`:initialValue | Configuración de la prop`, () => {
    it(`La prop tiene como valor por defecto 0 y es de tipo Number`, () => {
      const wrapper = shallowMount(Counter)
      const prop = wrapper.vm.$options.props.initialValue
      expect(prop.isRequired).toBeFalsy() // En nuestro caso no es obligatoria
      expect(prop.default).toBe(0) // Tiene 0 como valor por defecto
      expect(prop.type).toBe(Number) // Es de tipo Number
      expect(prop.validator(-10)).toBe(false) // No admite números negativos
    })
  })

})
