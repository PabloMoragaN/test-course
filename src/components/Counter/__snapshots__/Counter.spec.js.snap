// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter test suite :initialValue | Cuando se pasa 4 como valor inicial El contador incrementa su valor en 1 1`] = `"<p>4</p>"`;
